### DISCUSSION TITLE
Mise installed pythons are not detected by vscode

### DISCUSSION THREAD

**Discussion by ssbarnea** [hverlin/mise-vscode#71] [General]
While I am not sure if this extension can help vscode-python official extension to find python interpreters exposed by mise, it would clearly be very useful feature.

At this moment, even if mise is activated and working in vscode terminal, vscode python extension is failing to detect mise pythons and suggests only the ones installed via different methods. Should be noted that pyenv and asdf ones seems to be detected but I am no longer using those, since i switched to mise.

![](https://sbarnea.com/ss/Screen-Shot-2024-12-22-10-04-28.79.png)

Please remark that while the .venv is detected, the two mise installed pythons are not suggested by vscode when it comes to selecting a python interpreter, even if they are in PATH. This is an issue especially for projects that do not use a  `.venv`, and most python ones that use tox, do not have one as they rely on tox to manage these.

**Comment by hverlin** [comment]
I have documented the python + mise setup here: https://hverlin.github.io/mise-vscode/guides/python/

Let me know if this works for you. I suspect that you might run into an issue if you load it with multiple folders. In theory, your `settings.json` file should have been updated, but I don't see it on your screenshot.

**Comment by ssbarnea** [comment]
The problem is that vscode-python is unable to detect python interpreters from mise. On the other hand, if I have a `.venv` inside the project, it will find it. I updated the screenshot to explain the issue better.

**Comment by hverlin** [comment]
> The problem is that vscode-python is unable to detect python interpreters from mise
It does work if you set the default interpreter path in the `settings.json`

IMO, It's all caused by VSCode limitation on not having a good way to share settings (https://github.com/microsoft/vscode/issues/40233) so the python VSCode extension had to implement [some workarounds](https://github.com/microsoft/vscode-python/wiki/Setting-descriptions). 

If they would, the Python VSCode extension could simply let us update and read from `settings.json`. (For example, they could have an array where one could populate all the know python versions installed on the system). 

As you mentioned here https://github.com/microsoft/vscode-python/issues/20752#issuecomment-2558405412, the VSCode extension could also discover all python installations present in the `PATH`.

(note that there are workarounds for sharing settings currently, like https://marketplace.visualstudio.com/items?itemName=swellaby.workspace-config-plus, but best is to just do it manually. (you can add a mise task to merge the shared settings, for example)
